toolchain/base/value_store.h

```diff
@@ -113,9 +175,50 @@ class ValueStore
   }
 
  private:
+  // Poison the entire contents of the value store. This is used to detect cases
+  // where references to elements in a value store are used across calls that
+  // might modify the store.
+  [[gnu::always_inline]] auto PoisonAll() const -> void {
```

Comment:
Was the function call overhead here visible in fastbuild? I would have somewhat expected it to not be, but maybe I'm wrong...

Unless it's visible, I'd skip the always inline...

If it is visible, I'd suggest at least a comment on the first one explaining that, as it seems likely to be a surprise for anyone later.

Reply:
Did some quick measurements (ran all the tests 10x). Very noisy results. I'm not seeing any difference for `-c fastbuild`, but `-c dbg` seems to be around 1% slower without the attributes.